panels/user-accounts/cc-avatar-chooser.c

```diff
@@ -411,77 +411,133 @@ cheese_camera_device_monitor_new_cb (GObject *source,
-                if (enumerator == NULL) {
-                        g_object_unref (dir);
+
+                if (enumerator == NULL)
                         continue;
-                }
 
-                while ((info = g_file_enumerator_next_file (enumerator, NULL, NULL)) != NULL) {
-                        added_faces = TRUE;
+                while ((infoptr = g_file_enumerator_next_file (enumerator, NULL, NULL)) != NULL) {
+                        g_autoptr (GFileInfo) info = infoptr;
+                        g_autoptr (GFile) face_file = NULL;
 
                         type = g_file_info_get_file_type (info);
-                        if (type != G_FILE_TYPE_REGULAR &&
-                            type != G_FILE_TYPE_SYMBOLIC_LINK) {
-                                g_object_unref (info);
+                        if (type != G_FILE_TYPE_REGULAR && type != G_FILE_TYPE_SYMBOLIC_LINK)
                                 continue;
-                        }
 
                         target = g_file_info_get_symlink_target (info);
-                        if (target != NULL && g_str_has_prefix (target , "legacy/")) {
-                                g_object_unref (info);
+                        if (target != NULL && g_str_has_prefix (target , "legacy/"))
                                 continue;
-                        }
 
-                        file = g_file_get_child (dir, g_file_info_get_name (info));
-                        g_list_store_append (self->faces, file);
-
-                        g_object_unref (info);
+                        face_file = g_file_get_child (dir, g_file_info_get_name (info));
+                        g_list_store_append (faces, face_file);
+                        added_faces = TRUE;
                 }
 
                 g_file_enumerator_close (enumerator, NULL, NULL);
-                g_object_unref (enumerator);
-                g_object_unref (dir);
 
-                if (added_faces)
+                if (added_faces && !add_all)
                         break;
         }
+        return added_faces;
+}
+
+static void
+setup_photo_popup (CcAvatarChooser *self)
+{
+        g_auto(GStrv) facesdirs;
+        gboolean added_faces = FALSE;
+
+        self->faces = g_list_store_new (G_TYPE_FILE);
+        gtk_flow_box_bind_model (GTK_FLOW_BOX (self->flowbox),
+                                 G_LIST_MODEL (self->faces),
+                                 create_face_widget,
+                                 self,
+                                 NULL);
+
+        g_signal_connect (self->flowbox, "child-activated",
+                          G_CALLBACK (face_widget_activated), self);
+
+        facesdirs = get_settings_facesdirs ();
+        added_faces = add_faces_from_dirs (self->faces, facesdirs, TRUE);
+
+        if (!added_faces) {
+                facesdirs = get_system_facesdirs ();
+                add_faces_from_dirs (self->faces, facesdirs, FALSE);
```

Comment:
Any reason to not always include faces from system dirs instead of only when `added_faces` is false? Similarly, why not pass TRUE to `add_all` here? If that is the case we could remove the `add_all` param altogether from `add_faces_from_dirs`.

Reply:
Also clarified upstream.